compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1366,6 +1366,35 @@ impl SanitizerSet {
         })
     }
 
+    fn from_json(json: &Json) -> Result<Self, String> {
+        let mut sanitizer_set = SanitizerSet::empty();
+        if let Some(sanitizers) = json.as_array() {
+            for sanitizer in sanitizers {
+                let name = sanitizer
+                    .as_str()
+                    .ok_or_else(|| "Sanitizer name must be a string".to_string())?;
+                sanitizer_set |= match name {
```

Comment:
Same as above. Would you mind if we consider and follow up on this in a separate PR as part of/under #123619 when moving it to the rustc_sanitizers crate?

Reply:
Could we take care of this now? I would hate for us to accidentally accept `asan` in one place and `address` in the other just because these functions got out of sync with each other 🙂

Reply:
Done. Let me know if it looks good to you.

Reply:
Yes, that looks great to me, thank you!